### PR TITLE
[Internal] Benchmark tool: Fixes code refractoring to model the metrics as EventSource

### PR DESCRIPTION
**Pull request by kirankumarkolli**
# Pull Request Template

## Description

Benchamrk tool refactoring. Change metrics collection using the EventSource-based approach

image

Here is another example with throughput exceeded failures
image

## Type of change
Changed metrics collection using EventSource - EventListener approach
Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

**Review by ealsur — APPROVED** [review]

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/AzureVmBenchmark/scripts/custom-script.sh [hunk 1]
```diff
@@ -14,10 +14,10 @@ git checkout ${BENCHMARKING_TOOLS_BRANCH_NAME}
 cd 'Microsoft.Azure.Cosmos.Samples/Tools/Benchmark'
 
 echo "########## Build benckmark tool ##########"
-dotnet build --configuration Release -p:"OSSProjectRef=true;ShouldUnsetParentConfigurationAndPlatform=false"
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/AzureVmBenchmark/scripts/custom-script.sh:17]: Are these related to refractoring?

Can we please have them as separate PR?

Comment (tangerinous) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/AzureVmBenchmark/scripts/custom-script.sh]: done, that line was fixed in this PR
https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4066

**Review by kundadebdatta — APPROVED** [review]

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/AzureVmBenchmark/system/cloud-init.txt [hunk 2]
```diff
@@ -4,9 +4,6 @@ packages:
   - azure-cli
 
 runcmd:
-  - wget https://aka.ms/downloadazcopy-v10-linux
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/AzureVmBenchmark/system/cloud-init.txt:7]: Is it related to refractoring?

Comment (tangerinous) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/AzureVmBenchmark/system/cloud-init.txt:7]: This line is not related to refactoring; I added it by mistake in previous PR and believed that I could remove it using this PR

Comment (tangerinous) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/AzureVmBenchmark/system/cloud-init.txt:7]: we used this tool before Debdata offered us a better solution and I forgot to remove it
